In [24]:
from pymatgen import Lattice, Structure
from pymatgen.analysis.diffraction.xrd import XRDCalculator

from jupyter_jsmol.pymatgen import quick_view

In [25]:
struct = Structure.from_file("RbTlO2_mp-8176_symmetrized.cif")

In [44]:
Diffract = XRDCalculator()
xrd = Diffract.get_pattern(struct, two_theta_range=(0,30))
print(xrd, "\n")
print(xrd.hkls)

DiffractionPattern
$2\Theta$: [13.70056508 27.6020334  29.44708834]
Intensity: [100.          86.37155173  36.56553191] 

[[{'hkl': (0, 0, 0, 3), 'multiplicity': 2}], [{'hkl': (0, 0, 0, 6), 'multiplicity': 2}], [{'hkl': (1, 0, -1, 1), 'multiplicity': 12}]]


La radiation est celle d'une $CuK\alpha$ par défaut pour la classe XRDCalculator, il n'est donc pas nécessaire de la préciser. Les 3 premiers pics de diffraction se trouvent dans l'intervalle d'angle [0°;30°]. On a 13.7° correspondant au hkl=(0,0,3), 27.6° pour hkl=(0,0,6) et 29.4° pour hkl=(1,0,1).